signal_timeout_block.py

```diff
@@ -67,8 +68,10 @@ def start(self):
                         # scheduled before this persisted job is scheduled, so
                         # only do this if one has not been scheduled already.
                         self._schedule_timeout_job(job, key, interval, True)
+        self._persistence_scheduled.set()
```

Comment:
That is the approach I was initially going to to take but I thought the behavior could still be weird if signals are processed before the persisted jobs are scheduled. The way I made it is so that any new signals will cancel and re-schedule the persisted jobs.